lib/rules/no-duplicate-imports.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @fileoverview Restrict usage of duplicate imports.
+ * @author Simen Bekkhus
+ * @copyright 2016 Simen Bekkhus. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/**
+ * Returns the name of the module imported or re-exported.
+ *
+ * @param {ASTNode} node - A node to get.
+ * @returns {String|Boolean} the name of the module, or false if no name.
+ */
+function getValue(node) {
+    if (node && node.source && node.source.value) {
+        return node.source.value.trim();
+    }
+
+    return false;
+}
+
+/**
+ * Checks if the name of the import or export exists in the given array, and reports if so.
+ *
+ * @param {RuleContext} context - The ESLint rule context object.
+ * @param {ASTNode} node - A node to get.
+ * @param {String} value - The name of the imported or exported module.
+ * @param {Array} array - The array containing other imports or exports in the file.
+ * @param {String} message - A message to be reported after the name of the module
+ *
+ * @returns {undefined} No return value
+ */
+function checkAndReport(context, node, value, array, message) {
+    if (array.indexOf(value) !== -1) {
+        context.report({
+            node: node,
+            message: "'{{module}}' " + message,
+            data: {module: value}
+        });
+    }
+}
+
+/**
+ * Returns a function handling the imports of a given file
+ *
+ * @param {RuleContext} context - The ESLint rule context object.
+ * @param {Boolean} includeExports - Whether or not to check for exports in addition to imports.
+ * @param {Array} importsInFile - The array containing other imports in the file.
+ * @param {Array} exportsInFile - The array containing other exports in the file.
+ *
+ * @returns {Function} A function passed to ESLint to handle the statement.
+ */
+function handleImports(context, includeExports, importsInFile, exportsInFile) {
+    return function(node) {
+        var value = getValue(node);
+
+        if (value) {
+            checkAndReport(context, node, value, importsInFile, "import is duplicated.");
+
+            if (includeExports) {
+                checkAndReport(context, node, value, exportsInFile, "import is duplicated as export.");
+            }
+
+            importsInFile.push(value);
+        }
+    };
+}
+
+/**
+ * Returns a function handling the exports of a given file
+ *
+ * @param {RuleContext} context - The ESLint rule context object.
+ * @param {Array} importsInFile - The array containing other imports in the file.
+ * @param {Array} exportsInFile - The array containing other exports in the file.
+ *
+ * @returns {Function} A function passed to ESLint to handle the statement.
```

Comment:
Protip: Instead of just saying `{function}` you can actually define the function signature via `@callback` or `@typedef`: http://stackoverflow.com/a/17811177/90297